types/tables.d.ts

```diff
@@ -48,10 +48,10 @@ export interface ArcTable<Item = unknown> {
   query(params: QueryParams): Promise<QueryOutput<Item>>;
   query(params: QueryParams, callback: Callback<QueryOutput<Item>>): void;
 
-  scan(params: ScanParams): Promise<ScanOutput<Item>>;
-  scan(params: ScanParams, callback: Callback<ScanOutput<Item>>): void;
+  scan(params?: ScanParams): Promise<ScanOutput<Item>>;
+  scan(params?: ScanParams, callback: Callback<ScanOutput<Item>>): void;
```

Comment:
I don't think a required parameter can follow an optional parameter. I would probably split this out into two further function signatures.

I'll fork from your branch and send a new PR with an adjustment, to maintain your credit.